Jenkinsfiles/build.groovy

```diff
@@ -0,0 +1,258 @@
+    contextSource: [$class: 'ManuallyEnteredCommitContextSource', context: context],
+    statusResultSource: [$class: 'ConditionalStatusResultSource', results: [[$class: 'AnyBuildResult', message: message, state: state]]],
+  ])
+}
+
+String getCurrentVersion() {
+  return readMavenPom().getVersion()
+}
+
+String getNewVersion() {
+  String currentVersion = getCurrentVersion()
+  return BRANCH_NAME == 'master' ? currentVersion : "${BRANCH_NAME}-" + currentVersion
+}
+
+void runFunctionalTests(String baseDir) {
+  try {
+    sh "mvn -B -nsu -f ${baseDir}/pom.xml verify"
+  } finally {
+    try {
+      archiveArtifacts allowEmptyArchive: true, artifacts: "${baseDir}/**/target/failsafe-reports/*, ${baseDir}/**/target/**/*.log, ${baseDir}/**/target/*.png, ${baseDir}/**/target/**/distribution.properties, ${baseDir}/**/target/**/configuration.properties"
+    } catch (err) {
+      echo hudson.Functions.printThrowable(err)
+    }
+  }
+}
+
+void updateVersion() {
+  withEnv([
+    "CURRENT_VERSION=${getCurrentVersion()}",
+    "NEW_VERSION=${getNewVersion()}",
+  ]) {
+    echo """
+    ----------------------------------------
+    Update version
+    ----------------------------------------
+    Current version: ${CURRENT_VERSION}
+    New version: ${NEW_VERSION}
+    """
+    echo "MAVEN_OPTS=$MAVEN_OPTS"
+    sh """
+      mvn -nsu versions:set -DnewVersion=${NEW_VERSION} -DgenerateBackupPoms=false
+      perl -i -pe 's|<nuxeo.jsf.version>${CURRENT_VERSION}</nuxeo.jsf.version>|<nuxeo.jsf.version>${NEW_VERSION}</nuxeo.jsf.version>|' pom.xml
+    """
+  }
+}
+
+pipeline {
+  agent {
+    label 'jenkins-nuxeo-jsf-11'
+  }
+  environment {
+    MAVEN_OPTS= "$MAVEN_OPTS -Xms512m -Xmx3072m"
+  }
+
+  stages {
+    stage('Workaround bower root issue') {
+      steps {
+        container('maven') {
+          sh """
+            # to be removed when running with jenkins user
+            echo '{ "allow_root": true }' > /home/jenkins/.bowerrc
+          """
+        }
+      }
+    }
+
+    stage('Check Nuxeo branch') {
+      when {
+        branch 'PR-*'
+      }
+      steps {
+        container('maven') {
+          withCredentials([usernameColonPassword(credentialsId: 'jx-pipeline-git-github-git', variable: 'GITHUB_USERNAME_PASSWORD')]) {
+            script {
+              def branchRef = sh(script: "curl -u '${GITHUB_USERNAME_PASSWORD}' https://api.github.com/repos/nuxeo/nuxeo/git/ref/heads/${CHANGE_BRANCH} | jq -r '.ref'", returnStdout: true).trim();
+              echo "branch found: ${branchRef}"
+              if (branchRef != null) {
+                currentBuild.result = 'ABORTED';
+                currentBuild.description = "Branch: ${CHANGE_BRANCH} exists on nuxeo/nuxeo repository, aborting build."
+                error(currentBuild.description)
+              }
+            }
+          }
+        }
+      }
+    }
+
+    stage('Update version') {
+      steps {
+        container('maven') {
+          updateVersion()
```

Comment:
Any reason for this function? Used only once, and for the other stages, the code is directly inside, maybe it's clearer.

Reply:
Aligned with the `build-status.groovy` Jenkinsfile, so I may want to keep it as it.